templates/terraform/custom_expand/subnetwork_log_config.go.erb

```diff
@@ -13,13 +13,21 @@
 	# limitations under the License.
 -%>
 func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d TerraformResourceData, config *Config) (interface{}, error) {
+	purpose, ok := d.GetOkExists("purpose")
+
+	if ok && purpose.(string) == "INTERNAL_HTTPS_LOAD_BALANCER" {
+		// Subnetworks for L7ILB do not accept any values for logConfig
+		return nil, nil
+	}
+
 	l := v.([]interface{})
 	transformed := make(map[string]interface{})
 	if len(l) == 0 || l[0] == nil {
 		// send enable = false to ensure logging is disabled if there is no config
 		transformed["enable"] = false
 		return transformed, nil
```

Comment:
Mind adding that check here instead? We only want to return `nil` if we would set it to false. That way the API will fail requests with the block set for real.